unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarLOD/Tests/AvatarsLODControllerShould.cs

```diff
@@ -132,41 +131,134 @@ public void UnregisterPlayerProperly_NotRegistered()
+            CommonScriptableObjects.playerUnityPosition.Set(cameraPosition);
+
+            // Create full avatar to reach max 3D avatars
+            Player fullAvatarPlayer = CreateMockPlayer("fullAvatar");
+            IAvatarLODController fullAvatarPlayerController = Substitute.For<IAvatarLODController>();
+            fullAvatarPlayerController.player = fullAvatarPlayer;
+            controller.lodControllers.Add(fullAvatarPlayer.id, fullAvatarPlayerController);
+            fullAvatarPlayer.worldPosition = cameraPosition + Vector3.forward * (lodDistance * 0.1f);
+
+            // Create impostor avatar
+            Player impostorPlayer = CreateMockPlayer("impostorPlayer", out IAvatarRenderer impostorPlayerRenderer);
+            otherPlayers.Add(impostorPlayer.id, impostorPlayer);
+            var lodController = Substitute.For<IAvatarLODController>();
+            lodController.player = impostorPlayer;
+            controller.lodControllers.Add(impostorPlayer.id, lodController);
+            impostorPlayer.worldPosition = cameraPosition + Vector3.forward * lodDistance * 1.1f + Vector3.left * 3;
+
+            controller.enabled = true;
+            controller.Update();
+
+            impostorPlayerRenderer.Received().SetImpostorForward(-(cameraPosition - impostorPlayer.worldPosition).normalized);
         }
 
         [Test]
-        public void UpdateAllLODsCorrectly_Distance()
+        public void AvoidInvisiblesBillboardRotation()
         {
-            otherPlayers.Add("player0", CreateMockPlayer("player0", Vector3.forward * DataStore.i.avatarsLOD.simpleAvatarDistance.Get() * 0.25f)); //Close player => Full Avatar
-            otherPlayers.Add("player1", CreateMockPlayer("player1", Vector3.forward * DataStore.i.avatarsLOD.simpleAvatarDistance.Get() * 1.05f)); //Near By player => Simple Avatar
-            otherPlayers.Add("player2", CreateMockPlayer("player2", Vector3.forward * DataStore.i.avatarsLOD.LODDistance.Get() * 1.05f)); //Far Awaya player => LOD
-            controller.lodControllers.Add("player0", CreateMockLODController(otherPlayers["player0"]));
-            controller.lodControllers.Add("player1", CreateMockLODController(otherPlayers["player1"]));
-            controller.lodControllers.Add("player2", CreateMockLODController(otherPlayers["player2"]));
+            Vector3 cameraPosition = Vector3.zero;
+            float lodDistance = DataStore.i.avatarsLOD.LODDistance.Get();
+            CommonScriptableObjects.cameraForward.Set(Vector3.forward);
+            CommonScriptableObjects.cameraPosition.Set(cameraPosition);
+            CommonScriptableObjects.playerUnityPosition.Set(cameraPosition);
+
+            // Create full avatar to reach max 3D avatars
+            Player fullAvatarPlayer = CreateMockPlayer("fullAvatar");
+            IAvatarLODController fullAvatarPlayerController = Substitute.For<IAvatarLODController>();
+            fullAvatarPlayerController.player = fullAvatarPlayer;
```

Comment:
Let's use mock:
```
fullAvatarPlayerController.player.Returns(fullAvatarPlayer);
```

This also allows us to avoid the setter in `IAvatarLODController.player`.

Reply:
Awsome, I didn't know that, thx! I just applied that and reverted the public setter I had added.